modules/registry-refactoring/src/main/java/org/apache/airavata/registry/core/repositories/expcatalog/ExperimentSummaryRepository.java

```diff
@@ -0,0 +1,265 @@
+                    queryParameters.put(DBConstants.Experiment.GATEWAY_ID, filters.get(field));
+                    query += "ES.gatewayId LIKE :" + DBConstants.Experiment.GATEWAY_ID + " AND ";
+                }
+
+                else if (field.equals(DBConstants.Experiment.PROJECT_ID)) {
+                    logger.debug("Filter Experiments by Project ID");
+                    queryParameters.put(DBConstants.Experiment.PROJECT_ID, filters.get(field));
+                    query += "ES.projectId LIKE :" + DBConstants.Experiment.PROJECT_ID + " AND ";
+                }
+
+                else if (field.equals(DBConstants.Experiment.EXPERIMENT_NAME)) {
+                    logger.debug("Filter Experiments by Name");
+                    queryParameters.put(DBConstants.Experiment.EXPERIMENT_NAME, filters.get(field));
+                    query += "ES.name LIKE :" + DBConstants.Experiment.EXPERIMENT_NAME + " AND ";
+                }
+
+                else if (field.equals(DBConstants.Experiment.DESCRIPTION)) {
+                    logger.debug("Filter Experiments by Description");
+                    queryParameters.put(DBConstants.Experiment.DESCRIPTION, filters.get(field));
+                    query += "ES.description LIKE :" + DBConstants.Experiment.DESCRIPTION + " AND ";
+                }
+
+                else if (field.equals(DBConstants.Experiment.EXECUTION_ID)) {
+                    logger.debug("Filter Experiments by Execution ID");
+                    queryParameters.put(DBConstants.Experiment.EXECUTION_ID, filters.get(field));
+                    query += "ES.executionId LIKE :" + DBConstants.Experiment.EXECUTION_ID + " AND ";
+                }
+
+            }
+
+        }
+
+        if (filters.get(DBConstants.ExperimentStatus.STATE) != null) {
+            logger.debug("Filter Experiments by State");
+            String state = ExperimentState.valueOf(filters.get(DBConstants.ExperimentStatus.STATE)).toString();
+            queryParameters.put(DBConstants.ExperimentStatus.STATE, state);
+            query += "ES.state LIKE :" + DBConstants.ExperimentStatus.STATE + " AND ";
+        }
+
+        if (filters.get(DBConstants.ExperimentSummary.FROM_DATE) != null
+                && filters.get(DBConstants.ExperimentSummary.TO_DATE) != null) {
+
+            Timestamp fromDate = Timestamp.valueOf(filters.get(DBConstants.ExperimentSummary.FROM_DATE));
+            Timestamp toDate = Timestamp.valueOf(filters.get(DBConstants.ExperimentSummary.TO_DATE));
+
+            if (toDate.after(fromDate)) {
+                logger.debug("Filter Experiments by CreationTime");
+                queryParameters.put(DBConstants.ExperimentSummary.FROM_DATE, fromDate);
+                queryParameters.put(DBConstants.ExperimentSummary.TO_DATE, toDate);
+                query += "ES.creationTime BETWEEN :" + DBConstants.ExperimentSummary.FROM_DATE + " AND :" + DBConstants.ExperimentSummary.TO_DATE + " AND ";
+            }
```

Comment:
This should probably at least log an error if the fromDate comes after toDate.  Throwing an IllegalArgumentException would work too.